torch_lr_finder/lr_finder.py

```diff
@@ -338,3 +414,30 @@ def __del__(self):
         for k in self.cached:
             if os.path.exists(self.cached[k]):
                 os.remove(self.cached[k])
+
+
+class DataLoaderIterWrapper(object):
+    """
+    A wrapper for iterating `torch.utils.data.DataLoader` with the ability to reset
+    itself while `StopIteration` is raised.
+    """
+
+    def __init__(self, data_loader, auto_reset=True):
+        self.data_loader = data_loader
+        self.auto_reset = auto_reset
+        self._iterator = iter(data_loader)
+
+    def __next__(self):
+        # Get a new set of inputs and labels
+        try:
+            inputs, labels = next(self._iterator)
+        except StopIteration:
+            if not self.auto_reset:
+                raise
+            self._iterator = iter(self.data_loader)
+            inputs, labels = next(self._iterator)
+
+        return inputs, labels
+
+    def get_batch(self):
+        return next(self)
```

Comment:
For Python 2 support add the `next` function:
```
# For Python 2
next = __next__
```

Reply:
Got it! I did forget to test it on Python 2, my bad.